Перед вами результаты наблюдений длительности нахождения человека в очереди в зависимости от количества людей в этой очереди. Обучите модель линейной регрессии для прогнозирования и введите указанные параметры.

In [1]:
import pandas as pd
import numpy as np

In [2]:
data_loc = 'time_in_line.csv'

In [3]:
data = pd.read_csv(data_loc)

In [4]:
data.head()

id   X   Y
0   1  15  37
1   2  11  33
2   3   6  13
3   4  21  37
4   5  17  49

In [5]:
x_mean = data['X'].mean()

In [6]:
x_mean

12.9

In [7]:
y_mean = data['Y'].mean()

In [8]:
y_mean

28.0

In [9]:
n = len(data)

In [10]:
mean_sum = 0
mean_square_sum = 0

In [11]:
for i in range(n):
    mean_sum += (data['X'][i] - x_mean)*(data['Y'][i] - y_mean)
    mean_square_sum += (data['X'][i] - x_mean)**2  

In [12]:
theta1 = mean_sum/mean_square_sum

In [13]:
theta1

1.632435316084289

In [14]:
theta0 = y_mean - theta1*x_mean

In [15]:
theta0

6.941584422512669

In [16]:
sum_y_thetas_difference_squared = 0
ys_difference_squared = 0

In [17]:
for i in range(n):
    sum_y_thetas_difference_squared += (data['Y'][i] - theta0 - theta1*data['X'][i])**2
    ys_difference_squared += (data['Y'][i] - y_mean)**2

In [18]:
r_squared = 1 - sum_y_thetas_difference_squared/ys_difference_squared

In [19]:
r_squared

0.6607476279388789

Определите выборочное среднее X: 12.9

Определите выборочное среднее Y: 28

Найдите коэффициент θ_1: 1.63

Найдите коэффициент θ_0: 6.94

Оцените точность модели, вычислив R^2 статистику: 0.66